### PR TITLE
cloudinfo: identify gce from file

### PR DESCRIPTION
**Pull request by enoodle**
Fixes https://github.com/google/cadvisor/issues/1273

### CODE REVIEW

File: utils/cloudinfo/gce.go [hunk 1]
```diff
@@ -15,15 +15,25 @@
...
 	"strings"
 
 	info "github.com/google/cadvisor/info/v1"
 
 	"google.golang.org/cloud/compute/metadata"
 )
 
+const (
+	ProductNameFileName = "/sys/class/dmi/id/product_name"
+	Google              = "Google"
```
Comment (timstclair) [inline utils/cloudinfo/gce.go]: nit: these don't need to be exported

Comment (timstclair) [inline utils/cloudinfo/gce.go]: nit: call this `gceProductName`

File: utils/cloudinfo/gce.go [hunk 2]
```diff
@@ -15,15 +15,25 @@
...
+)
+
 func onGCE() bool {
-	return metadata.OnGCE()
+	data, err := ioutil.ReadFile(ProductNameFileName)
+	if err != nil {
```
Comment (timstclair) [inline utils/cloudinfo/gce.go]: Let's log the error at info (`V(2)`)

**Comment by timstclair** [comment]
ok to test

**Comment by enoodle** [comment]
@timstclair I fixed your comments.

**Comment by timstclair** [comment]
LGTM, but needs a rebase.

**Comment by timstclair** [comment]
ok to test

**Comment by vishh** [comment]
@k8s-bot test this

**Comment by timstclair** [comment]
ok to test

**Comment by timstclair** [comment]
Sorry, needs another rebase.

**Comment by enoodle** [comment]
rebased

**Pull request merged**